pep-0629.rst

```diff
@@ -0,0 +1,167 @@
+PEP: 629
+Title: Remove wstr from Unicode
+Author: Inada Naoki <songofacandy@gmail.com>
+Status: Draft
+Type: Standards Track
+Content-Type: text/x-rst
+Created: 25-Jun-2020
+Python-Version: 3.10~3.12
+
+
+Abstract
+========
+
+PEP 393 deprecated some unicode APIs, and introduced ``wchar_t *wstr``,
+and ``Py_ssize_t wstr_length`` in Unicode implementation to support
+these deprecated APIs. [1]_
+
+This PEP is planning removal of ``wstr``, and ``wstr_length`` with
+deprecated APIs using these members by Python 3.12.
```

Comment:
> Most APIs are documented as deprecated since Python 3.3, and marked Py_DEPRECATED since Python 3.6. Most of them can be removed in 3.10.

I'm not sure that people are paying attention to compiler warnings emitted by Py_DEPRECATED(). Do you think that it's worth it to start emiting a DeprecationWarning in Python 3.10, and remove these functions in Python 3.12, or maybe already in Python 3.11?

Reply:
I don't think so.

For example, PyLong_FromUnicode doesn't execute arbitrary code.  It doesn't release GIL too.
If we emit warning from the function, now the function will run arbitrary code (warning filter).

It is breaking change too. But compile error must be safe than runtime warning.
We can revert the removal anyway.